public-ui/src/app/data-browser/components/source-tree/source-tree-react.component.tsx

```diff
@@ -59,32 +59,21 @@ export const SourceTreeComponent = (
             super(props);
             this.state = {
                 children: undefined,
-                isConceptSelected: false,
                 isHandelSelected: undefined,
-                highlightId: undefined
             };
         }
         conceptClick() {
-            localStorage.setItem('treeHighlight', this.props.node.id);
-            this.setState({
-                isConceptSelected: !this.state.isConceptSelected
-            });
-            this.getTreeHighlight();
-        }
-        getTreeHighlight() {
-            this.setState({
-                highlightId: parseInt(localStorage.getItem('treeHighlight'), 10)
-            });
+            localStorage.setItem('selectedTreeNode', JSON.stringify(this.props.node));
```

Comment:
This is where the data should bubble up to the parent. Local Storage doesn't seem like the best solution.